### ISSUE TITLE
Allow custom app_offline.htm in web app deployment task

### ISSUE THREAD

**Issue by martincostello** [Microsoft/vsts-tasks#3425]
From looking at the [code](https://github.com/Microsoft/vsts-tasks/blob/588387ac2fb3caf73c6f7e1f0f1eec069b6363f7/Tasks/Common/webdeployment-common/kuduutility.ts#L14) it appears that the ```app_offline.htm``` file is generated from a hard-coded string.

It would be nice if there was a way to specify a custom ```app_offline.htm``` file path and/or content as a parameter to the task to allow for a nicer UX to any user that is served the offline page as a deployment is being performed.

**Comment by kirkone** [comment]
@martincostello had the same problem.
I did some work on my own Build Tasks. If you want please have a look here: [vsts-dnx-tasks](https://github.com/kirkone/vsts-dnx-tasks)
You can either use the default App_Offline.htm which looks like other Azure status sites or you can provide a custom page.

I would appreciate some feedback

Greetings, KirK

**Comment by RoopeshNair** [comment]
We have added this to our backlog. However it will be 2-3 sprints before we can get to it.
As we are tracking this on our backlog, closing the issue
Thanks,
Roopesh Nair

**Comment by kirkone** [comment]
Hi,

@RoopeshNair Would it be helpful to submit a pull request for this?
I am working on something similar and it would be me an honor to be able to contribute something.
Greetings,
KirK

**Comment by johnkrussell** [comment]
@RoopeshNair Did this ever make it into a sprint? I've been looking for a solution but I'm unable to find one, and it's something that would be really useful.

**Comment by RoopeshNair** [comment]
@johnkrussell, not yet. We are working on few high priority items that are impacting large number of customers. It will be a while before we can get to it.

We do encourage contributions, if you are open to it, please submit a PR 👍

**Comment by StephanBis** [comment]
What's the status on this, has it been implemented yet?

**Comment by kirkone** [comment]
Hi, I lost this a bit out of scope but @StephanBis reminded me to get my PR ready. I will try to get this as a first draft in the next few days and come back here.

**Comment by fncap** [comment]
Any news on this issue?

**Comment by kedigby** [comment]
Any news on this?

**Comment by RoopeshNair** [comment]
Unfortunately, our team was unable to prioritize it this as they are working on problems with broader customer impact. We encourage contributions - @kirkone :)

**Comment by kirkone** [comment]
Totally lost track on this. I will have a look at this again. Will not be in the next few days but as soon as I can.

**Comment by nhojt** [comment]
Any news on this?

**Comment by DennisJansenDev** [comment]
It's been 5 years now since it was put on the backlog, any news on this?

**Comment by Walkman100** [comment]
Took me a couple days, but I figured out how to do this fairly simply: https://stackoverflow.com/a/77964681/2999220

Adding `app_offline.htm` works to provide a custom page while publishing... but in my experience doesn't get deleted. I adapted the `PublishWebPackage.ps1` script by kirkone linked in the first comment to be able to delete it afterwards - PS modules used to get the website details (Get-AzureWebsite) don't work anymore, they have been superseded by Get-AzWebApp which doesn't provide the same details.
(@DennisJansenDev 7 years now...)
